app/js/filterFunctions.ts

```diff
@@ -122,17 +123,14 @@ function percentageFilter(resultArray: Array<any>) {
  */
 function testAllocatedFilter(resultArray: Array<any>) {
     let filterTestAllocated = document.querySelector<HTMLInputElement>('#testAllocated');
-    let newResultArray = resultArray;
     let chosenTestAllocated = filterTestAllocated.value;
     if (chosenTestAllocated) {
-        newResultArray = [];
-        resultArray.forEach((data)=>{
-            if(data.testId === chosenTestAllocated) {
-                newResultArray.push(data)
-            }
-        });
+        let newResultArray = resultArray.filter((data) => {
```

Comment:
you can just return `resultArray.filter...` instead of putting it in a variable and returning the variable